EngineHacks/SkillSystem/Skills/PreBattleSkills/Loyalty/Loyalty.s

```diff
@@ -32,28 +31,21 @@ cmp r2,#0x0
 beq End
 add r0,#0x1
 
+@	Make Loyalty check for Lord bit instead of ClassID
 mov r3,#0x48
 ldr r5,CharData
 sub r2,#0x1
 mul r3,r2
 add r5,r3
-mov r3,#0x4
-ldr r5,[r5,r3]
-ldrb r3,[r5,r3]
-cmp r3,#0x1
-beq Final
-cmp r3,#0x2
-beq Final
-cmp r3,#0x3
-beq Final
-cmp r3,#0x4
-beq Final
-cmp r3,#0x3B
-beq Final
-cmp r3,#0x3C
-beq Final
-b Loop
-
+ldr r0, [r5] @char
+ldr r0, [r0, #0x28] @char abilities
+ldr r1, [r5,#4] @class
+ldr r1, [r1,#0x28] @class abilities
+orr r0, r1
+mov r1, #0x20
+lsl r1, #8
+tst r0, r1
+beq End
```

Comment:
This code doesn't loop through the buffer `GetUnitsInRange`  returns, so a lord unit can be missed if they aren't the first adjacent unit found by `GetUnitsInRange`.

Looping through the buffer to check every found allied unit in range should fix this.